.github/workflows/build-and-push.yml

```
@@ -0,0 +1,37 @@
+
+env:
+  REGISTRY: docker.io
+  IMAGE_NAME: appwrite/mailcatcher
+  TAG: ${{ github.event.release.tag_name }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v3
```

actionlint (1.7.8)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)